src/SprykerEco/Zed/Payone/Business/Payment/MethodSender/PayoneGetInvoiceMethodSender.php

```diff
@@ -0,0 +1,129 @@
+use SprykerEco\Zed\Payone\Business\Payment\DataMapper\StandartParameterMapperInterface;
+use SprykerEco\Zed\Payone\Business\Payment\PaymentMapperReader;
+use SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface;
+
+class PayoneGetInvoiceMethodSender implements PayoneGetInvoiceMethodSenderInterface
+{
+    public const ERROR_ACCESS_DENIED_MESSAGE = 'Access denied';
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Api\Adapter\AdapterInterface
+     */
+    protected $executionAdapter;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface
+     */
+    protected $queryContainer;
+
+    /**
+     * @var \Generated\Shared\Transfer\PayoneStandardParameterTransfer
+     */
+    protected $standardParameter;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Payment\PaymentMethodMapperInterface[]
+     */
+    protected $registeredMethodMappers;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Payment\DataMapper\StandartParameterMapperInterface
+     */
+    protected $standartParameterMapper;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Payment\PaymentMapperReader
+     */
+    protected $paymentMapperManager;
+
+    /**
+     * @param \SprykerEco\Zed\Payone\Business\Api\Adapter\AdapterInterface $executionAdapter
+     * @param \SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface $queryContainer
+     * @param \Generated\Shared\Transfer\PayoneStandardParameterTransfer $standardParameter
+     * @param \SprykerEco\Zed\Payone\Business\Payment\DataMapper\StandartParameterMapperInterface $standartParameterMapper
+     * @param \SprykerEco\Zed\Payone\Business\Payment\PaymentMapperReader $paymentMapperReader
+     */
+    public function __construct(
+        AdapterInterface $executionAdapter,
+        PayoneQueryContainerInterface $queryContainer,
+        PayoneStandardParameterTransfer $standardParameter,
+        StandartParameterMapperInterface $standartParameterMapper,
+        PaymentMapperReader $paymentMapperReader
+    ) {
+        $this->executionAdapter = $executionAdapter;
+        $this->queryContainer = $queryContainer;
+        $this->standardParameter = $standardParameter;
+        $this->standartParameterMapper = $standartParameterMapper;
+        $this->paymentMapperReader = $paymentMapperReader;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PayoneGetInvoiceTransfer $getInvoiceTransfer
+     *
+     * @return \Generated\Shared\Transfer\PayoneGetInvoiceTransfer
+     */
+    public function getInvoice(PayoneGetInvoiceTransfer $getInvoiceTransfer): PayoneGetInvoiceTransfer
+    {
+        $responseContainer = new GetInvoiceResponseContainer();
+        $paymentEntity = $this->findPaymentByInvoiceTitleAndCustomerId(
+            $getInvoiceTransfer->getReference(),
+            $getInvoiceTransfer->getCustomerId()
+        );
+
+        if ($paymentEntity) {
```

Comment:
* Extract to a method `fetch Invoice(PayoneGetInvoiceTransfer):Abstract ResponseContainer`. 
* Get rid from `else` condition.

Reply:
++